Server/src/index.ts

```diff
@@ -3,169 +3,319 @@
+    this.initAjv(ajv);
 
-        process.on('SIGINT', this.shutdown.bind(this));
-        process.on('SIGTERM', this.shutdown.bind(this));
-        process.on('SIGQUIT', this.shutdown.bind(this));
-    }
+    // Initialize parent logger
+    this.initLogger();
 
-    protected _createSender(): IMessageSender {
-        return new RabbitMqSender(this._config, this._logger);
-    }
+    // Force sync database
+    this.forceDbSync();
+
+    // Set cache implementation
+    this.initCache(cache);
+
+    // Initialize Swagger if enabled
+    this.initSwagger();
+
+    // Register AJV for schema validation
+    this.registerAjv();
+
+    // Initialize module & API
+    // Always initialize API after SwaggerUI
+    this.initSystem(appName);
+
+    process.on('SIGINT', this.shutdown.bind(this))
+    process.on('SIGTERM', this.shutdown.bind(this))
+    process.on('SIGQUIT', this.shutdown.bind(this))
+  }
+
+  private initHealthCheck() {
+    this._server.get('/health', async () => {
+      return {status: 'healthy'}
+    })
+  }
+
+  private initAjv(ajv?: Ajv) {
+    this._ajv = ajv || new Ajv({
+      removeAdditional: 'all',
+      useDefaults: true,
+      coerceTypes: 'array',
+      strict: false
+    })
+    addFormats(this._ajv, {
+      mode: 'fast',
+      formats: ['date-time']
+    })
+  }
+
+  private initLogger() {
+    this._logger = new Logger<ILogObj>({
+      name: 'CitrineOS Logger',
+      minLevel: systemConfig.server.logLevel,
+      hideLogPositionForProduction: systemConfig.env === 'production'
+    })
+  }
+
+  private forceDbSync() {
+    sequelize.DefaultSequelizeInstance.getInstance(this._config, this._logger, true)
+  }
 
-    protected _createHandler(): IMessageHandler {
-        return new RabbitMqReceiver(this._config, this._logger);
+  private initCache(cache?: ICache) {
+    this._cache = cache || (this._config.util.cache.redis
+      ? new RedisCache({
+        socket: {
+          host: this._config.util.cache.redis.host,
+          port: this._config.util.cache.redis.port
+        }
+      })
+      : new MemoryCache())
+  }
+
+  private initSwagger() {
+    if (this._config.util.swagger) {
+      initSwagger(this._config, this._server)
     }
+  }
 
-    shutdown() {
+  private registerAjv() {
+    // todo type schema instead of any
+    const fastifySchemaCompiler: FastifySchemaCompiler<any> = (routeSchema: FastifyRouteSchemaDef<any>) => {
+      return this._ajv?.compile(routeSchema.schema) as FastifyValidationResult;
+    };
+    this._server.setValidatorCompiler(fastifySchemaCompiler);
+  }
 
-        // Shut down all modules and central system
-        this._modules.forEach(module => {
-            module.shutdown();
-        });
-        this._centralSystem.shutdown();
+  private initCentralSystem() {
+    this._centralSystem = new CentralSystemImpl(this._config, this._cache as ICache, undefined, undefined, this._logger, this._ajv)
+    this.host = this._config.server.host
```

Comment:
i'm receiving this additional error while trying to get the docker running:

`Server/src/index.ts(174,30): error TS2339: Property 'server' does not exist on type '{ env: "development" | "production"; data: { sequelize: { host: string; port: number; database: string; username: string; password: string; storage: string; sync: boolean; dialect?: any; }; }; centralSystem: { ...; }; ... 4 more ...; maxCachingSeconds: number; }'`

i locally updated this to be `this.host = this._config.centralSystem.host` and `this.port = this._config.centralSystem.port` and the docker containers ran successfully - should this same change be made to the PR as well or is this unique to my machine?

Reply:
@venus-yu should be fixed